Fast Pair: align after sysbuild migration (#16022)
nrfconnect/sdk-nrf: doc/nrf/external_comp/bt_fast_pair.rst

The Fast Pair standard requires provisioning the device with Model ID and Anti-Spoofing Private Key obtained during device model registration.
In the |NCS|, the provisioning data is generated as a hexadecimal file using the :ref:`bt_fast_pair_provision_script`.

.. _ug_bt_fast_pair_provisioning_script_triggers:

> Build system triggers for the provision script

Reviewer: I would suggest to simplify this a bit and state that:
```
Build system automatically calls the script and includes the resulting hexadecimal file in the firmware (the :file:`merged.hex` file) if application enables Fast Pair as described in :ref:`_ug_bt_fast_pair_prerequisite_ops`.
You must provide the following additional CMake options to build the application enabling Fast Pair:
```

The Fast Pair standard requires provisioning the device with Model ID and Anti-Spoofing Private Key obtained during device model registration.
In the |NCS|, the provisioning data is generated as a hexadecimal file using the :ref:`bt_fast_pair_provision_script`.

> The build system automatically calls the Fast Pair provision script and includes the resulting hexadecimal file in the firmware (the :file:`merged.hex` file) when building with Fast Pair in the |NCS|.

Reviewer 2: ```suggestion
When building the Fast Pair in the |NCS|, the build system automatically calls the Fast Pair provision script and includes the resulting hexadecimal file in the firmware (the :file:`merged.hex` file).
```

> You must provide the following additional CMake options to build an application with Fast Pair support:

Reviewer 2: ```suggestion
To build an application with the Fast Pair support, include the following additional CMake options:
```

For example, when building an application with the |nRFVSC|, you need to add the following parameters in the **Extra CMake arguments** field on the **Add Build Configuration view**: ``-DFP_MODEL_ID=0xFFFFFF -DFP_ANTI_SPOOFING_KEY=AbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbA=``.
Make sure to replace ``0xFFFFFF`` and ``AbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbA=`` with values obtained for your device.
See :ref:`cmake_options` for more information about defining CMake options.

> See the following sections for information on how to add the Google Fast Pair subsystem to your project.

Author: Maybe we should be more specific here, now that we have an explicit link

```suggestion
See the following :ref:`ug_bt_fast_pair_prerequisite_ops_kconfig` section for information on how to add the Google Fast Pair subsystem to your project.
```

Reviewer: I meant not only Kconfig, but whole interation (including APIs etc.)

> See the following sections for information on how to add the Google Fast Pair subsystem to your project.

Author: Could you please use this sentence directly after the sentence from L177?

_The build system automatically calls the Fast Pair provision script and includes the resulting hexadecimal file in the firmware (the :file:`merged.hex` file) when building with Fast Pair in the |NCS|._

Performing prerequisite operations
**********************************

> 

Author: You need to also write one additional sentence before creating subsections. E.g.

`To support the Google Fast Pair subsystem in your project, you need to follow the steps in the following subsections that describe necessary Kconfig and API actions.`

> To add the Google Fast Pair subsystem to your project, you must enable the ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option if you use the default |NCS| build system configuration with sysbuild.

Author: Could you create a dedicated section: `Kconfig` with a dedicated link to this section? _ug_bt_fast_pair_prerequisite_ops still seems broad

If you do not use sysbuild, you must enable :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option at the main application image level.

.. note::
   Sysbuild copies the value of the ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option into the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is part of the main application image.
   Your configuration of the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option at the main application image level has no effect as this Kconfig option is directly set by sysbuild.

> Fast Pair API integration

Author: ```suggestion
API
```
?

This part does not describe the whole Fast Pair API integration

Or longer names

Performing prerequisite Kconfig operations
Performing prerequisite API operations

Reviewer: You are right that naming may be a bit misleading here (e.g. FMDN APIs are not covered here)

Performing prerequisite operations
**********************************

> To support the Google Fast Pair subsystem in your project, you need to follow the steps in the following subsections that describe necessary Kconfig and API actions:

Reviewer 2: ```suggestion
To integrate the Google Fast Pair subsystem in your project, complete the following steps:
```

Author: adding word prerequisite as these steps are not sufficient for the whole integration (we have other steps in the integration guide):

```
To start integrating the Google Fast Pair subsystem in your project, complete the following prerequisite steps:
```

* :ref:`ug_bt_fast_pair_prerequisite_ops_kconfig`
* :ref:`ug_bt_fast_pair_prerequisite_ops_api`

> The subsequent subsections describe prerequisite operations for Fast Pair extensions supported in the |NCS|.

Reviewer 2: ```suggestion
The subsequent subsections describe requires steps for enabling Fast Pair extensions supported in the |NCS|.
```
Felt it's missing a word - not sure if the meaning is maintained

> Remaining subsections describe prerequisite operations for Fast Pair extensions supported in the |NCS|.

Reviewer 2: ```suggestion
The subsequent subsections describe prerequisite operations for Fast Pair extensions supported in the |NCS|.
```

.. _ug_bt_fast_pair_prerequisite_ops_kconfig:

Enabling Fast Pair in Kconfig
=============================

> To add the Google Fast Pair subsystem to your project, you must enable the ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option if you use the default |NCS| build system configuration with sysbuild.

Reviewer 2: ```suggestion
If you are using the default |NCS| build system configuration with sysbuild and wish to add the Google Fast Pair subsystem to your project, enable the ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option.
```
for better flow

If you do not use sysbuild, you must enable :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option at the main application image level.

.. note::

>    Sysbuild copies the value of the ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option into the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option that is part of the main application image.
>    Your configuration of the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option at the main application image level has no effect as this Kconfig option is directly set by sysbuild.

Reviewer 2: ```suggestion
   Sysbuild sets the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option in the main application image based on the value of the ``SB_CONFIG_BT_FAST_PAIR`` Kconfig option. 
   Your configuration of the :kconfig:option:`CONFIG_BT_FAST_PAIR` Kconfig option at the main application image will be ineffective, as sysbuild overrides it.
```

****************

The :ref:`bt_fast_pair_provision_script` is required to generate the provisioning data for the device.

> The script is automatically called by the build system in typical scenarios.

Reviewer: `when building with Fast Pair in the |NCS|.`

****************

The :ref:`bt_fast_pair_provision_script` is required to generate the provisioning data for the device.

> The script is automatically called by the build system when building with Fast Pair in the |NCS|.

Reviewer 2: ```suggestion
The build system calls it automatically when building with Fast Pair in the |NCS|.
```
